scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -58,6 +58,13 @@ pub struct TomlManifest {
     pub tool: Option<BTreeMap<SmolStr, MaybeWorkspaceTomlTool>>,
     pub features: Option<BTreeMap<FeatureName, Vec<TomlFeatureToEnable>>>,
     pub patch: Option<BTreeMap<SmolStr, BTreeMap<PackageName, TomlDependency>>>,
+    pub target_defaults: Option<BTreeMap<TargetKind, TargetDefaults>>,
```

Comment:
This is good, but I think we should also allow inheriting this setting from ws, so all of these:

```Scarb.toml
[package] 
name = "hello"

[target-defaults.test]
build-external-contracts = [ "..." ]
```

and 

```Scarb.toml
[package] 
name = "hello"

[workspace.target-defaults.test]
build-external-contracts = [ "..." ]

[target-defaults]
test.workspace = true
```

and 

```Scarb.toml
[package] 
name = "hello"

[workspace.target-defaults.test]
build-external-contracts = [ "..." ]

[target-defaults.test]
build-external-contracts.workspace = true
```

would be valid (and equivalent). 

Can be a separate PR imho.

Reply:
Ok, will do 👌